pkg/util/validation/limits.go

```diff
@@ -106,15 +109,25 @@ func IsLimitError(err error) bool {
 // limits via flags, or per-user limits via yaml config.
 type Limits struct {
 	// Distributor enforced limits.
-	RequestRate                                 float64             `yaml:"request_rate" json:"request_rate"`
-	RequestBurstSize                            int                 `yaml:"request_burst_size" json:"request_burst_size"`
-	IngestionRate                               float64             `yaml:"ingestion_rate" json:"ingestion_rate"`
-	IngestionBurstSize                          int                 `yaml:"ingestion_burst_size" json:"ingestion_burst_size"`
-	IngestionBurstFactor                        float64             `yaml:"ingestion_burst_factor" json:"ingestion_burst_factor" category:"experimental"`
-	AcceptHASamples                             bool                `yaml:"accept_ha_samples" json:"accept_ha_samples"`
-	HAClusterLabel                              string              `yaml:"ha_cluster_label" json:"ha_cluster_label"`
-	HAReplicaLabel                              string              `yaml:"ha_replica_label" json:"ha_replica_label"`
-	HAMaxClusters                               int                 `yaml:"ha_max_clusters" json:"ha_max_clusters"`
+	RequestRate          float64 `yaml:"request_rate" json:"request_rate"`
+	RequestBurstSize     int     `yaml:"request_burst_size" json:"request_burst_size"`
+	IngestionRate        float64 `yaml:"ingestion_rate" json:"ingestion_rate"`
+	IngestionBurstSize   int     `yaml:"ingestion_burst_size" json:"ingestion_burst_size"`
+	IngestionBurstFactor float64 `yaml:"ingestion_burst_factor" json:"ingestion_burst_factor" category:"experimental"`
+	AcceptHASamples      bool    `yaml:"accept_ha_samples" json:"accept_ha_samples"`
+	HAClusterLabel       string  `yaml:"ha_cluster_label" json:"ha_cluster_label"`
+	HAReplicaLabel       string  `yaml:"ha_replica_label" json:"ha_replica_label"`
+	HAMaxClusters        int     `yaml:"ha_max_clusters" json:"ha_max_clusters"`
+	// We should only update the timestamp if the difference
+	// between the stored timestamp and the time we received a sample at
+	// is more than this duration.
+	HATrackerUpdateTimeout          model.Duration `yaml:"ha_tracker_update_timeout" json:"ha_tracker_update_timeout" category:"advanced" doc:"flag=distributor.ha-tracker.update-timeout|description=Update the timestamp in the KV store for a given cluster/replica only after this amount of time has passed since the current stored timestamp."`
```

Comment:
Added a new `doc.flag` parameter to be able to override in docs which CLI flag populates this parameter, since we do it with custom code (`migrateParameter`).

Reply:
No longer necessary after [2ab77ef](https://github.com/grafana/mimir/pull/11774/commits/2ab77eff772829566cb16468afbb0d43747561ca).